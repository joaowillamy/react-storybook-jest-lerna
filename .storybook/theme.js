import { create } from "@storybook/theming/create";

export default create({
  base: "light",

  colorPrimary: "#FDCB13",
  colorSecondary: "#18ACC4",

  // UI
  appBg: "white",
  appContentBg: "#F2F3F4",
  appBorderColor: "#F2F3F4",
  appBorderRadius: 4,

  // Typography
  fontBase: '"Montserrat", sans-serif',
  fontCode: "monospace",

  brandTitle: "Quero front test"
  // brandUrl: 'http://resolvvi.com.br/',
  // brandImage: ''
});
