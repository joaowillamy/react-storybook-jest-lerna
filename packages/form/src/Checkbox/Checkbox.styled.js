import styled, { css } from "styled-components";

export const Checkbox = styled.div`
  display: inline;
`;

export const CheckboxLabel = styled.label`
  display: flex;
  align-items: center;
  cursor: pointer;
  height: auto;
  width: auto;
  outline: none;
  transition: all 0.3s;
`;

export const CheckboxContainer = styled.div`
  margin-right: 5px;
`;

export const CheckboxInput = styled.input`
  clip: rect(0 0 0 0);
  clippath: inset(50%);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  white-space: nowrap;
  width: 1px;
`;

export const CheckboxIcon = styled.div`
  background: ${({ isChecked, theme }) =>
    isChecked ? theme.colors.bluePrimary : "#fff"};
  border-radius: 5px;
  display: flex;
  flex-direction: column;
  align-items: center;
  height: 20px;
  transition: all 150ms;
  width: 20px;
  border: 0.5px solid
    ${({ theme, isChecked }) =>
      isChecked ? theme.colors.bluePrimary : theme.colors.black};

  svg {
    margin-bottom: 1px;
    color: #fff;
    visibility: ${({ isChecked }) => (isChecked ? "visible" : "hidden")};
  }
`;
