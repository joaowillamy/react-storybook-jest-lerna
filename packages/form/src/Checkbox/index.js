import React, { useState, useCallback } from "react";
import PropTypes from "prop-types";
import { Icon } from "@joaowillamy-test-quero/core";

import * as S from "./Checkbox.styled";

const Checkbox = ({ label, isSelected, value, name, onChange }) => {
  const [isChecked, setIsChecked] = useState(isSelected || false);
  const onChangeCallback = useCallback(onChange);

  const toggleIsChecked = () => {
    setIsChecked(!isChecked);
    onChangeCallback &&
      onChangeCallback({ value, name, isChecked: !isChecked });
  };

  return (
    <S.Checkbox>
      <S.CheckboxLabel>
        <S.CheckboxContainer>
          <S.CheckboxInput
            type="checkbox"
            value={value}
            name={name}
            checked={isChecked}
            onChange={toggleIsChecked}
          />
          <S.CheckboxIcon isChecked={isChecked}>
            <Icon.Check />
          </S.CheckboxIcon>
        </S.CheckboxContainer>
        {label}
      </S.CheckboxLabel>
    </S.Checkbox>
  );
};

Checkbox.propTypes = {
  name: PropTypes.string,
  value: PropTypes.any,
  onChange: PropTypes.func,
  isSelected: PropTypes.bool
};

export default Checkbox;
