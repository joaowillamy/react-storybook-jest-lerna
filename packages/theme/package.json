{
  "name": "@will-ui-kit/theme",
  "version": "1.0.2",
  "main": "lib/index.js",
  "module": "src/index.js"
}
