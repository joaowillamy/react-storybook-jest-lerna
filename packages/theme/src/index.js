export { default as ThemeProvider } from "./ThemeProvider";
export { default as DEFAULT_THEME } from "./theme";
export { GlobalStyles } from "./GlobalStyles";
