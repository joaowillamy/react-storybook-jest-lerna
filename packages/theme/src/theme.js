const BASE = 3;

const borders = {
  sm: color => `1px solid ${color || "#00000029"}`,
  md: color => `3px solid ${color || "#00000029"}`
};

const colors = {
  bluePrimary: "#18ACC4",
  blueSecondary: "#007A8D",
  yellowPrimary: "#FDCB13",
  yellowSecondary: "#DE9E1F",
  green: "#0FA866",
  black: "#1F2D30",
  grey: "#FBFBFB",
  Overlay: "#1F2D30e0" // com 88% de opacidade.
};

const fonts = {
  Montserrat: "Montserrat"
};

const fontSizes = {
  xs: "12px",
  sm: "14px",
  md: "16px",
  lg: "18px",
  xl: "22px",
  xxl: "26px",
  xxxl: "36px"
};

const fontWeights = {
  thin: 100,
  extralight: 200,
  light: 300,
  regular: 400,
  medium: 500,
  semibold: 600,
  bold: 700,
  extrabold: 800,
  black: 900
};

const iconSizes = {
  sm: "12px",
  md: "16px",
  lg: "26px"
};

const shadows = {
  sm: color => `0 1px 3px ${color || "#00000029"}`,
  md: color => `0 3px 4px ${color || "#00000029"}`,
  lg: color => `0 3px 6px ${color || "#00000029"}`
};

const space = {
  xs: `${BASE}px`,
  sm: `${BASE * 2}px`,
  md: `${BASE * 3}px`,
  lg: `${BASE * 4}px`,
  xl: `${BASE * 8}px`,
  xxl: `${BASE * 10}px`,
  xxxl: `${BASE * 12}px`
};

const DEFAULT_THEME = {
  borders,
  colors,
  fonts,
  fontSizes,
  fontWeights,
  iconSizes,
  shadows,
  space
};

export default DEFAULT_THEME;
