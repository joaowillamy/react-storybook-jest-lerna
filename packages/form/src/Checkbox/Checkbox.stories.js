import React from "react";
import { ThemeProvider } from "@joaowillamy-test-quero/theme";
import { Card } from "@joaowillamy-test-quero/core";

import Checkbox from ".";

export default {
  component: Checkbox,
  title: "form|Checkbox"
};

export const checkbox = () => {
  return (
    <ThemeProvider>
      <div style={{ width: "100px", display: "inline-block" }}>
        <Checkbox
          label={<>oi</>}
          name={"checkbox"}
          value={{ a: 2, b: [1, 2] }}
          onChange={data => {
            console.log(data);
          }}
        />
      </div>
      <div style={{ width: "100px", display: "inline-block" }}>
        <Checkbox
          label={<>oi</>}
          name={"checkbox"}
          isSelected={true}
          value={{ a: 2, b: [1, 2] }}
          onChange={data => {
            console.log(data);
          }}
        />
      </div>
    </ThemeProvider>
  );
};

export const checkboxElements = () => {
  return (
    <ThemeProvider>
      <Checkbox
        label={
          <>
            <Card />
          </>
        }
        name={"checkbox"}
        value={{ a: 2, b: [1, 2] }}
        onChange={data => {
          console.log(data);
        }}
      />
    </ThemeProvider>
  );
};
