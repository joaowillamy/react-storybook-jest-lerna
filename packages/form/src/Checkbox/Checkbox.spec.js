import React from "react";
import renderer from "react-test-renderer";
import "jest-styled-components";
import { ThemeProvider } from "@joaowillamy-test-quero/theme";

import Checkbox from ".";

describe("Checkbox", () => {
  test("Checkbox", () => {
    const tree = renderer
      .create(
        <ThemeProvider>
          <Checkbox />
        </ThemeProvider>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
