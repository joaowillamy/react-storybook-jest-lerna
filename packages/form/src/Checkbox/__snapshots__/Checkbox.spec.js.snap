// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox Checkbox 1`] = `
.c5 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c0 {
  display: inline;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  height: auto;
  width: auto;
  outline: none;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
}

.c2 {
  margin-right: 5px;
}

.c3 {
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  -webkit-clippath: inset(50%);
  clippath: inset(50%);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  white-space: nowrap;
  width: 1px;
}

.c4 {
  background: #fff;
  border-radius: 5px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 20px;
  -webkit-transition: all 150ms;
  transition: all 150ms;
  width: 20px;
  border: 0.5px solid #1F2D30;
}

.c4 svg {
  margin-bottom: 1px;
  color: #fff;
  visibility: hidden;
}

<div
  className="c0"
>
  <label
    className="c1"
  >
    <div
      className="c2"
    >
      <input
        checked={false}
        className="c3"
        onChange={[Function]}
        type="checkbox"
      />
      <div
        className="c4"
      >
        <svg
          aria-hidden="true"
          className="c5"
          fill="currentColor"
          focusable="false"
          viewBox="0 0 16 16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M10.97 4.97a.75.75 0 011.071 1.05l-3.992 4.99a.75.75 0 01-1.08.02L4.324 8.384a.75.75 0 111.06-1.06l2.094 2.093 3.473-4.425a.236.236 0 01.02-.022z"
            fillRule="evenodd"
          />
        </svg>
      </div>
    </div>
  </label>
</div>
`;
